modules/parachains/src/lib.rs

```diff
@@ -467,28 +576,21 @@ pub mod pallet {
 				Error::<T, I>::HeaderChainStorageProof(HeaderChainError::StorageProof(e))
 			})?;
 
-			Ok(PostDispatchInfo { actual_weight: Some(actual_weight), pays_fee: Pays::Yes })
-		}
-
-		/// Change `PalletOwner`.
-		///
-		/// May only be called either by root, or by `PalletOwner`.
-		#[pallet::call_index(1)]
-		#[pallet::weight((T::DbWeight::get().reads_writes(1, 1), DispatchClass::Operational))]
-		pub fn set_owner(origin: OriginFor<T>, new_owner: Option<T::AccountId>) -> DispatchResult {
-			<Self as OwnedBridgeModule<_>>::set_owner(origin, new_owner)
-		}
+			// check if we allow this submission for free
+			let is_free = total_parachains == 1
```

Comment:
We better go the other way instead :) I.e. to change the pallet to only track one parachain head. When this pallet was implemented, we thought it may be used to track multiple pallets, but actually we don't need that. We even had the https://github.com/paritytech/parity-bridges-common/issues/1869 to change it to track single pallet only, but it got closed. So right now our transaction extensions, relayer and runtime configurations are aligned - they're all configured to always submit/accept single parachain head.

Reply:
I agree 👍 I reopened #1869 and renamed it to `"Downgrade" parachains pallet to only support tracking single remote parachain (BridgeHub)`